### PR TITLE
Temporarily pin RMM while refactor removes deprecated calls

### PR DESCRIPTION
**Pull request by dantegd**
This should unblock build locally and in CI. Needs to be undone after #1707 is merged. 

cc @ChuckHastings

### CODE REVIEW

**Review by BradReesWork — APPROVED** [review]

**Comment by BradReesWork** [comment]
rerun tests

**Comment by dantegd** [comment]
@BradReesWork just realized that the RAFT PR https://github.com/rapidsai/raft/pull/315 needs to be merged first so that this one passes CI

File: cpp/cmake/thirdparty/get_rmm.cmake
Comment (rlratzel) [inline cpp/cmake/thirdparty/get_rmm.cmake:36]: Can a `FIXME` be added somewhere so we remember to set `GIT_SHALLOW` back to `TRUE` when we eventually go back to using a branch name again?  I think we'll want this to minimize the cloning overhead by not getting the entire git history when set back to `TRUE`.

**Review by rlratzel — APPROVED** [review]
Thanks!  One minor request if possible, which need not hold up approval.

**Review by rlratzel — APPROVED** [review]
Thanks for updating the conda files too, that should cover everything now.

**Comment by BradReesWork** [comment]
@gpucibot merge

**Comment by BradReesWork** [comment]
@gpucibot merge

**Review by ajschmidt8 — APPROVED** [review]

**Pull request merged**
